### PR TITLE
Fix temp dir creation

### LINKED ISSUE #2017: Bug: SDK binary creates a bunch of spam folders in /tmp
Moved from there: https://github.com/cosmos/cosmos-sdk/issues/22385.

I have this thing on Neutron testnet node, and I saw people on Discord reporting this as well.

So, somehow, there are hundreds of thousands (for me it was like 1.3m) of folders being created under /tmp, with the same naming pattern: `/tmp/<binary name><some random number, probably block height>`.
These folders are never deleted, eventually taking all of the free space of the server, even if the data folder of the node itself is on another volume.

Example on Neutron testnet:
```
validator@pion-testnet /tmp ❯ ls -laU | head -n 20                                                                                                                                        ✔ PIPE|0 12:04:50
total 2732664
drwxrwxrwt    1 root      root      59850752 Oct 27 12:04 .
drwxr-xr-x   18 root      root          4096 Jun 24 07:23 ..
drwxr-xr-x    4 validator validator     4096 Oct 23 16:40 neutrond553039366
drwxr-xr-x    4 validator validator     4096 Oct 23 14:44 neutrond2004005568
drwxr-xr-x    4 validator validator     4096 Oct 23 02:31 neutrond331565302
drwxr-xr-x    4 validator validator     4096 Oct 23 01:26 neutrond3645941473
drwxr-xr-x    4 validator validator     4096 Oct 24 21:24 neutrond2105281424
drwxr-xr-x    4 validator validator     4096 Oct 21 23:21 neutrond2013767621
drwxr-xr-x    4 validator validator     4096 Oct 23 10:12 neutrond2808888439
drwxr-xr-x    4 validator validator     4096 Oct 24 06:39 neutrond435430921
drwxr-xr-x    4 validator validator     4096 Oct 26 02:01 neutrond1138501452
drwxr-xr-x    4 validator validator     4096 Oct 21 06:02 neutrond2960233487
drwxr-xr-x    4 validator validator     4096 Oct 19 23:40 neutrond1006881956
drwxr-xr-x    4 validator validator     4096 Oct 26 17:16 neutrond2270539921
drwxr-xr-x    4 validator validator     4096 Oct 23 13:47 neutrond626593812
drwxr-xr-x    4 validator validator     4096 Oct 25 17:10 neutrond3787200811
drwxr-xr-x    4 validator validator     4096 Oct 20 20:20 neutrond2127120401
drwxr-xr-x    4 validator validator     4096 Oct 22 10:33 neutrond3242892381
drwxr-xr-x    4 validator validator     4096 Oct 25 12:15 neutrond693181263

validator@pion-testnet /tmp ❯ ls -1q | wc -l                                                                                                                                                       12:03:31
661652
```

SDK guys said it might be wasm related, as I have only wasm files within these folders:
```
|-- neutrond3895874916
|   |-- data
|   `-- wasm
|       `-- wasm
|           |-- cache
|           |   `-- modules
|           |-- exclusive.lock
|           `-- state
|               `-- wasm
|-- neutrond397461145
|   |-- data
|   `-- wasm
|       `-- wasm
|           |-- cache
|           |   `-- modules
|           |-- exclusive.lock
|           `-- state
|               `-- wasm
|-- neutrond4073770021
|   |-- data
|   `-- wasm
|       `-- wasm
|           |-- cache
|           |   `-- modules
|           |-- exclusive.lock
|           `-- state
|               `-- wasm
```

### PR DESCRIPTION
**Pull request by chipshort**
fixes #2017

Looks like most chains copied this code for their wasmd integration.
We should probably inform them about this change.

### CODE REVIEW

File: cmd/wasmd/commands.go [hunk 1]
```diff
@@ -260,7 +260,6 @@ var tempDir = func() string {
 	if err != nil {
 		panic("failed to create temp dir: " + err.Error())
 	}
-	defer os.RemoveAll(dir)
```
Comment (chipshort) [inline cmd/wasmd/commands.go:263]: This defer call just immediately removed the folder after being created. This causes wasmvm to create the folder again and then it is never deleted. Over time that leads to a lot of temp folders accumulating when people run the cli.
What we actually want to do is to remove it after we are done with it.

**Review by pinosu — APPROVED** [review]
LGTM! 👍 
As you said, we need to communicate it to the chains. 
Maybe we could add a comment explaining why we need it etc. WDYT?

**Comment by chipshort** [comment]
Sounds good. I added a comment.

**Review by pinosu — APPROVED** [review]
Nice! LGTM 👍

**Pull request merged**